### DISCUSSION TITLE
Undefined Reference to Matplot++ Library Functions While Linking Custom Target

### DISCUSSION THREAD

**Discussion by mmtaksuu** [alandefreitas/matplotplusplus#155] [Q&A] [answered]
Hi, I am trying to integrate matplot++ library using Clion IDE by using MinGW32 GCC compiler.
I created CMakeList.txt file as described in the example files, then  Clion found Matplot++ library. But when I try to run my cpp project I have received 'undefined reference to `matplot::gca()' error while linking my project. 
How can I solve this issue. Is there any way to deal with this problem?

![image](https://user-images.githubusercontent.com/18046031/120081455-75f54d00-c0c6-11eb-9e88-2da5e80d2734.png)

Here is my CMakeList.txt file 
```cmake
cmake_minimum_required(VERSION 3.19)
project(matplot_simple)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_PREFIX_PATH     "C:/Program Files/matplotplusplus 1.1.0/lib/cmake")
set(CMAKE_MODULE_PATH     "C:/Program Files/matplotplusplus 1.1.0/lib/cmake/Matplot++")

add_executable(${PROJECT_NAME} main.cpp)

find_package(Filesystem REQUIRED)
find_package(Matplot++ REQUIRED)
target_link_libraries(${PROJECT_NAME} PUBLIC std::filesystem)
target_link_libraries(${PROJECT_NAME} PUBLIC Matplot++::matplot)
target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
```

Here is my code
```c++
#include <vector>
#include <matplot/matplot.h>

int main()
{
    std::vector<double> x = matplot::linspace(0, 2 * matplot::pi);
    std::vector<double> y = matplot::transform(x, [](auto x) { return sin(x); });
    matplot::plot(x, y, "-o");
    matplot::show();

    return 0;
}
```

**Comment by alandefreitas** [comment, marked as answer]
It seems like you installed Matplot++ with MSVC, and the MSVC binaries are not compatible with MinGW32.

MinGW32 is not officially supported but you can try to [build from source](https://alandefreitas.github.io/matplotplusplus/integration/install/build-from-source/build-and-install/) with MinGW32 or use [FetchContent](https://alandefreitas.github.io/matplotplusplus/integration/cmake/embed-with-automatic-download/).

This *might* work.

If this doesn't work, you move to MSVC (CLion works with MSVC now) or you can adapt the build script to work with MinGW (and even open a PR if that works :D).

**Reply by mmtaksuu** [reply]
Hi,  I really thank you for your answer. I fixed my building problem by using GWSL and Ubuntu WSL 2 distro on Windows 10 machine. It's so cool that using linux machine while working on Windows. I just build the library from source and use it. Thanks again.

**Reply by alandefreitas** [reply]
There was a recent PR to make sure the build script works for MinGW. You should be able to build it from source with both MSVC or MinGW now. We are going to include MinGW binaries in the next release.

**Reply by mmtaksuu** [reply]
It will be much better for windows users who are not develop at Visual Studio. Thank you for your interest 👍
